src/frontend/src/components/LinkEditView.tsx

```diff
@@ -0,0 +1,70 @@
+import { useAllLinkTypes } from '../hooks/projects.hooks';
+import LoadingIndicator from './LoadingIndicator';
+import ErrorPage from '../pages/ErrorPage';
+import { Button, Grid, IconButton, MenuItem, TextField } from '@mui/material';
+import { FieldArrayWithId, UseFieldArrayAppend, UseFieldArrayRemove, UseFormRegister } from 'react-hook-form';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { LinkCreateArgs } from 'shared';
+
+const LinkEditView: React.FC<{
+  ls: FieldArrayWithId[];
+  register: UseFormRegister<{
+    name: string;
+    budget: number;
+    summary: string;
+    crId: string;
+    rules: { rule: string }[];
+    links: LinkCreateArgs[];
+    goals: { bulletId: number; detail: string }[];
+    features: { bulletId: number; detail: string }[];
+    constraints: { bulletId: number; detail: string }[];
+  }>;
+  append: UseFieldArrayAppend<any, any>;
+  remove: UseFieldArrayRemove;
+  links: LinkCreateArgs[];
+}> = ({ ls, register, append, remove, links }) => {
+  const { isLoading, isError, error, data } = useAllLinkTypes();
+  if (isLoading || !data) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error.message} />;
+
+  return (
+    <>
+      {ls.map((_element, i) => {
```

Comment:
checking whether something is required by checking the index is WAY too jank. I think adding the required field on linkType would make this easier, you could just check that to see what should be enabled/disabled